src/player/PlayerEquipment.gd

```diff
@@ -31,9 +34,34 @@ func _ready() -> void:
 func get_inventory() -> Inventory:
 	return _inventory;
 
+
+func _on_inventory_inventory_event(item_stack, cursur_position, ev) -> void:
+	if ev is InputEventMouseButton && ev.pressed && ev.button_index == MOUSE_BUTTON_LEFT:
+		var hovered_rect: Rect2i;
+		
+		if item_stack == null:
+			hovered_rect = Rect2i(cursur_position, Vector2i.ONE);
+		else:
+			hovered_rect = Rect2i(item_stack.position, item_stack.get_rotated_size());
+		
+		var selection: InventoryRenderer.ItemSelection = _inventory_renderer.get_selection(ID_SELECTION)
+		if selection != null and selection.get_rect() == Rect2i(hovered_rect.position, hovered_rect.size):
+			_inventory_renderer.clear_selection(ID_SELECTION);
+		elif item_stack == null:
+			_inventory_renderer.clear_selection(ID_SELECTION);
+		else:
+			var color: Color = get_theme_color("selected_color", "InventoryRenderer");
+			var outline: Color = get_theme_color("selected_outline", "InventoryRenderer");
+			
```

Comment:
the selected color and outline should be moved out of the renderer theme and into a constant for this script